extensions/wikia/PortableInfobox/controllers/PortableInfoboxBuilderSpecialController.class.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Class WikiaMapsSpecialController
+ * @desc Special:Maps controller
+ */
+class PortableInfoboxBuilderSpecialController extends WikiaSpecialPageController {
+	const PAGE_NAME = 'PortableInfoboxBuilder';
+	const PAGE_RESTRICTION = 'editinterface';
+	const INFOBOX_BUILDER_MERCURY_ROUTE = 'infoboxBuilder';
+
+	/**
+	 * Special page constructor
+	 *
+	 * @param null $name
+	 * @param string $restriction
+	 * @param bool $listed
+	 * @param bool $function
+	 * @param string $file
+	 * @param bool $includable
+	 */
+	public function __construct( $name = null, $restriction = '', $listed = true, $function = false, $file = 'default', $includable = false ) {
+		parent::__construct( self::PAGE_NAME, self::PAGE_RESTRICTION, $listed, $function, $file, $includable );
+	}
+
+	/**
+	 * Infobox Builder special page
+	 */
+	public function index() {
+		$this->wg->SuppressPageHeader = true;
+		$this->wg->out->setHTMLTitle( wfMessage( 'portable-infobox-builder-tite' )->text() );
+
+		$title = $this->getTemplateTitle( $this->getPar() );
+		$noTemplateSet = empty( $title ) ? true : false;
+
+		if ( $noTemplateSet ) {
+			$this->response->setVal('noTemplateSet', true );
+			$this->response->setVal('setTemplateNameCallToAction', wfMessage(
+				'portable-infobox-builder-no-template-title-set' )->text() );
+		} else {
+			$url = $this->wg->server . '/' . self::INFOBOX_BUILDER_MERCURY_ROUTE . '/' . $title;
```

Comment:
Should we urlencode the $title?

I'm also thinking if something along the lines of
$url = implode( PATH_SEPARATOR, [$this->wg->server, self::INFOBOX_BUILDER_MERCURY_ROUTE, $title] );
wouldn't read a little nicer (but that's a personal preference)